### ISSUE TITLE
Uncaught TypeError: Cannot match against 'undefined' or 'null'

### ISSUE THREAD

**Issue by Ben3eeE** [atom/teletype#357]
1. Attempt to join a portal that no longer exists (e.g., `atom://teletype/portal/0d6bffc9-fafc-433f-8ec8-bbcd8ffcac15`)
1. While attempting to join, click the share button

**Gif of the guest window**

![teletype exception](https://user-images.githubusercontent.com/1058982/35582721-de8f3118-05ef-11e8-8006-5e3c073ee998.gif)

**Atom**: 1.26.0-beta0 x64
**Electron**: 1.7.11
**OS**: Microsoft Windows 10 Pro
**Thrown From**: [teletype](https://github.com/atom/teletype) package 0.12.0

### Stack Trace

Uncaught TypeError: Cannot match against 'undefined' or 'null'.

```
At C:\Users\bene\.atom\packages\teletype\lib\participants-component.js:39

TypeError: Cannot match against 'undefined' or 'null'.
    at ParticipantsComponent.renderParticipant (/packages/teletype/lib/participants-component.js:39:21)
    at ParticipantsComponent.render (/packages/teletype/lib/participants-component.js:28:37)
    at Object.initialize (/packages/teletype/node_modules/etch/lib/component-helpers.js:32:31)
    at /packages/teletype/lib/participants-component.js:9:10)
    at render (/packages/teletype/node_modules/etch/lib/render.js:17:25)
    at addChildren (/packages/teletype/node_modules/etch/lib/render.js:39:24)
    at render (/packages/teletype/node_modules/etch/lib/render.js:29:21)
    at Object.initialize (/packages/teletype/node_modules/etch/lib/component-helpers.js:40:23)
    at /packages/teletype/lib/guest-portal-binding-component.js:10:10)
    at render (/packages/teletype/node_modules/etch/lib/render.js:17:25)
    at updateChildren (/packages/teletype/node_modules/etch/lib/patch.js:123:34)
    at patch (/packages/teletype/node_modules/etch/lib/patch.js:18:9)
    at updateChildren (/packages/teletype/node_modules/etch/lib/patch.js:77:7)
    at patch (/packages/teletype/node_modules/etch/lib/patch.js:18:9)
    at updateSync (/packages/teletype/node_modules/etch/lib/component-helpers.js:116:20)
    at /packages/teletype/node_modules/etch/lib/component-helpers.js:72:7
    at DefaultScheduler.performUpdates (/packages/teletype/node_modules/etch/lib/default-scheduler.js:48:34)
```

### Commands

```
     -7:05.5.0 core:confirm (input.hidden-input)
     -6:59.7.0 find-and-replace:show (div.results-view.focusable-panel)
  2x -6:57.6.0 core:confirm (input.hidden-input)
     -6:38.2.0 project-find:show (input.hidden-input)
     -6:36.7.0 find-and-replace:show (input.hidden-input)
     -6:34.8.0 project-find:show (input.hidden-input)
     -6:33.8.0 core:confirm (input.hidden-input)
     -6:26.3.0 project-find:show (input.hidden-input)
     -6:24.2.0 find-and-replace:show (input.hidden-input)
     -6:15.4.0 core:confirm (input.hidden-input)
     -6:14.5.0 project-find:show (input.hidden-input)
     -6:12.7.0 core:confirm (input.hidden-input)
     -1:24.3.0 application:new-window (input.hidden-input)
     -0:01.3.0 application:new-file (input.hidden-input)
     -0:00.9.0 core:paste (input.hidden-input)
     -0:00.3.0 core:undo (input.hidden-input)
```

### Non-Core Packages

```
language-latex 1.1.1 
latex 0.49.0 
teletype 0.12.0 
```

**Comment by jasonrudolph** [comment]
@Ben3eeE: Thanks for figuring out how to reproduce this problem, and thank you for opening this issue. :bow:

**Comment by jasonrudolph** [comment]
We've found slightly shorter repro steps, so I've updated the issue body accordingly.

I'll capture the original steps here for posterity:

> 1. Open host/guest windows
> 1. Close the host window so the guest disconnects
> 1. In the guest window attempt to rejoin the portal. The connection will timeout
> 1. While rejoining click the `Share` button

**Comment by sguthals** [comment]
👋 I received this error today:
<img width="487" alt="screen shot 2018-04-09 at 8 33 46 am" src="https://user-images.githubusercontent.com/1314285/38509101-d4766680-3bd5-11e8-83d8-c747b4bf2b1d.png">
And "View Issue" brought me to #233 

I am running Teletype 0.12.0

I did the following:
1. Open my portal
2. Guests joined
3. Error ocurred

Not sure if the host window lost connection, but I didn't close it. I (as the host) received the error.

**Comment by jasonrudolph** [comment]
> 👋 I received this error today:

Thanks, @sguthals. :bow: Your experience sounds different than this particular issue, so I don't think our fix for this issue in dca9a99 would have helped the behavior you experienced. 😓 

It's possible that the root cause of #233 is still hiding in the codebase somewhere, and that our [potential fix for #233](https://github.com/atom/teletype/issues/233#issuecomment-378409910) didn't squash the problem after all. I've opened #360 to track the issue you experienced. If you run into it again, or discover any other steps that might help us reproduce it reliably, please leave a comment in #360 to let us know.
